application/src/ext/java/org/opentripplanner/ext/flex/FlexTripsMapper.java

```diff
@@ -54,6 +52,15 @@ public class FlexTripsMapper {
           trip.getId()
         );
         // result.add(new ContinuousPickupDropOffTrip(trip, stopTimes));
+      } else if (
+        stopTimes.size() < 2 &&
+        stopTimes.stream().anyMatch(st -> st.hasFlexWindow() || st.hasFlexibleStop())
```

Comment:
What is the need for this check? Can these stoptimes be any stop times or are they filtered somewhere to be just "flex" stop times? Should this check be moved to be the first if so you don't have to account for this edge case in the other checks?

Reply:
These are all stop times, not just the flex ones therefore you need something like that.

If I took your comment literally, then I can move the longish if statement to the top and that would look like that: https://github.com/leonardehrenfried/OpenTripPlanner/commit/ddf302a215752da222f59582aee10338

However, I don't find this any better.

Thinking a bit bigger, we could have a first step where I extract only the flex stop times and in a second one we sort it into the two types of flex trips and add the issues.